zerver/lib/streams.py

```diff
@@ -140,6 +139,25 @@ def send_stream_creation_event(
     send_event_on_commit(realm, event, user_ids)
 
 
+def get_stream_permission_default_group(setting_name: str, realm: Realm) -> UserGroup:
```

Comment:
I guess we should add a commit to use `get_role_based_system_groups_dict` to get all system groups in a single query in this PR itself.